harness/determined/keras/_tf_keras_trial.py

```diff
@@ -363,15 +363,17 @@ def compile_model(
         env: det.EnvContext,
         hvd_config: horovod.HorovodContext,
     ) -> None:
-        (
-            context.model,
-            compile_args.arguments["optimizer"],
-        ) = keras._get_multi_gpu_model_and_optimizer(
-            pre_compiled_model=context.model,
-            optimizer=compile_args.arguments["optimizer"],
-            env=env,
-            hvd_config=hvd_config,
-        )
+        assert context.model is not None
+        if hvd_config.use:
+            # Horovod doesn't know how to handle string-based optimizers.
+            if isinstance(compile_args.arguments["optimizer"], str):
+                raise det.errors.InvalidExperimentException("string optimizers are not supported")
+
+            compile_args.arguments["optimizer"] = hvd.DistributedOptimizer(
+                compile_args.arguments["optimizer"],
+                aggregation_frequency=hvd_config.aggregation_frequency,
+                average_aggregated_gradients=hvd_config.average_aggregated_gradients,
+            )
```

Comment:
can you update `support_determined_native` to False and i think there are some e2e tests we may need to delete